synfig-core/src/synfig/rendering/renderer.h

```diff
@@ -180,6 +180,7 @@ class Renderer: public etl::shared_object
 	static void deinitialize();
 	static void register_renderer(const String &name, const Renderer::Handle &renderer);
 	static void unregister_renderer(const String &name);
+        static void print_renderers();
```

Comment:
use TAB to indent

Reply:
Anyway, I think `Renderer` should not have this method.
Implement as `show_targets` does.

Reply:
I will have a look at it.

Reply:
@rodolforg ,
How should I implement it as `show_targets` does ?
Because Renderer doesn't have any method similar to `book()` ?

Reply:
Use the method Renderer::get_renderers()

Reply:
But the thing is we have to print the available renderers many times, such as that's why I did it that way.

https://github.com/synfig/synfig/blob/26402ef46bc851f79685f99af8d65c37ddc4fc31/synfig-core/src/tool/optionsprocessor.cpp#L706-L707

https://github.com/synfig/synfig/blob/26402ef46bc851f79685f99af8d65c37ddc4fc31/synfig-core/src/tool/optionsprocessor.cpp#L489

https://github.com/synfig/synfig/blob/26402ef46bc851f79685f99af8d65c37ddc4fc31/synfig-studio/src/player/main.cpp#L141

https://github.com/synfig/synfig/blob/26402ef46bc851f79685f99af8d65c37ddc4fc31/synfig-studio/src/player/main.cpp#L120

I agree that Renderer is not the best place to have the print_renderers method, but I don't know where I should put it.

Reply:
I think the main idea here is to return a list of renderers instead of just printing them out and doing whatever you want with them (print, add them to list, etc.). This will make this code more universal.

Reply:
Ok, then I will completely remove the print_renderers() method and just write for loops wherever there is requirement to print the renderers.